hermes-console/static/js/console/subscription/SubscriptionController.js

```diff
@@ -57,21 +78,16 @@ subscriptions.controller('SubscriptionController', ['SubscriptionRepository', 'S
             $scope.health = health;
         });
 
-        subscriptionRepository.lastUndelivered(topicName, subscriptionName).$promise
-                .then(function (lastUndelivered) {
-                    $scope.lastUndelivered = lastUndelivered;
-                })
-                .catch(function () {
-                    $scope.lastUndelivered = null;
-                });
+        $scope.$watch('userHasSufficientPrivileges', function(currentValue, oldValue) {
+            // user login + has privileges
+            if (currentValue === true && oldValue === false) {
+                getUndelivered();
+                getLastUndelivered();
+            }
+        });
 
-        subscriptionRepository.undelivered(topicName, subscriptionName).$promise
-            .then(function (undelivered) {
-                $scope.undelivered = undelivered;
-            })
-            .catch(function () {
-                $scope.undelivered = [];
-            });
+        getUndelivered();
+        getLastUndelivered();
```

Comment:
Given that there is the above watch, do we need these calls?

Reply:
I added this watch to make sure the messages are fetched when the user status changes to authenticated. But you are right, and after changing the if condition, we no longer need these calls.